appinventor/appengine/src/com/google/appinventor/client/boxes/PropertiesBox.java

```diff
@@ -38,5 +56,93 @@ private PropertiesBox() {
         false,  // startMinimized
         false,  // bodyPadding
         false); // highlightCaption
+
+    // UIBinder calls the private constructor, presumably through reflection, which
+    // disrupts the singleton pattern. This might not be the best way to deal with it.
+    INSTANCE = this;
+  }
+
+  /*
+   * Show the given component's properties in the properties panel.
+   */
+  public void show(YaFormEditor formEditor, boolean selected) {
+    MockForm form = formEditor.getForm();
+    List<MockComponent> components = form.getSelectedComponents();
```

Comment:
Rather than have to tie this specifically to YaFormEditor, it might make sense to have the function take the `List<MockComponent>` as an argument instead. This would help us when we implement alternative interfaces that may not directly involve a YaFormEditor (see also AlexaSkillDesigner in the Alexa sources).

Reply:
This is part of my phase 2 changes.

Reply:
I'd like to revisit this once you're in the office.

Reply:
Deferring to a future update...